R/summarize_weighted_production.R

```diff
@@ -86,40 +86,56 @@ summarize_weighted_metric <- function(data,
 }
 
 add_weighted_loan_percent_change <- function(data, use_credit_limit = FALSE) {
-  add_weighted_loan_metric(data, use_credit_limit, percent_change = TRUE)
+  add_weighted_loan_metric(data, use_credit_limit, metric = "percent_change")
 }
 
 add_weighted_loan_production <- function(data, use_credit_limit = FALSE) {
-  add_weighted_loan_metric(data, use_credit_limit, percent_change = FALSE)
+  add_weighted_loan_metric(data, use_credit_limit, metric = "production")
 }
 
-add_weighted_loan_metric <- function(data, use_credit_limit, percent_change) {
+add_weighted_loan_emission_factor <- function(data, use_credit_limit = FALSE) {
+  add_weighted_loan_metric(data, use_credit_limit, metric = "emission_factor")
+}
+
+add_weighted_loan_metric <- function(data, use_credit_limit, metric) {
```

Comment:
Consider enumerating options to `metric` 
-- https://principles.tidyverse.org/def-enum.html